tests/std/tests/Dev09_126254_persistent_aux_allocators/test.cpp

```diff
@@ -98,6 +98,13 @@ void dump_map() {
     }
 }
 
+void test_gh_2362() {
+    allocator<int> al;
+    int* ptr = al.allocate(0);
+    assert(ptr == nullptr);

```

Comment:
I'm not sure if this `assert` is needed. It is the current implementation detail, which can be freely changed.

Reply:
Should I add additional `al.deallocate(nullptr, 0);` or if we can change that `nullptr` is not returned for `allocate(0)` then `deallocate(nullptr, 0);` might be invalid too?

Reply:
I think the later, so just remove the assert.